### ISSUE TITLE
Calculate Accessibility Matrix Failed - RuntimeError: Parallelization using parallel_calculate_locations.py failed.

### ISSUE THREAD

**Issue by AuroraBao** [Esri/public-transit-tools#192]
Hi, there are some issue when executing the 'Calculate Accessibility Matrix' tool, "RuntimeError: Parallelization using parallel_calculate_locations.py failed."

I have checked and validated the inputs, network, and my computer cores all meet the requirement, it looks the Parallel processing code is not running correctly in my default Python 3 ArcGIS Environment, I am on ArcGIS Pro 3.1.1

Is this a known issue of this tool or could you please advise where it doesn't look right on my setting to run this tool? Thank you!

Please find the error details in the attached image:

![GithubMatrixToolIssue](https://github.com/Esri/public-transit-tools/assets/109064648/4e3d1e22-dc83-4735-bfc2-bc8f42eff692)

**Comment by mmorang** [comment]
Hi @AuroraBao.  I'm sorry you've experienced this problem, and unfortunately I can't reproduce it or figure out what's causing it.

The error returned by the tool is a non-specific error that occurs whenever something horrible occurs in the subprocess that precalculates network locations.  Normally that subprocess will return a more specific error, but in this case something so awful happened that it wasn't even able to do that.  Unfortunately, this means I really don't know what it was.

Here are a few things you can try to help you move forward with your analysis:
1. If you haven't already, make sure you have the latest tool code.  I haven't updated anything in this tool very recently, and the line numbers in your error message all match up, so I think this is unlikely to be the issue.
2. You could try updating your ArcGIS Pro software to the latest release (3.2), although I wasn't able to reproduce the problem in 3.1.1 either, and this tool should work fine in 3.1.1.
3. Turn off the option to precalculate network locations.  This means the analysis portion of the tool will run slower, but you can also manually run the Calculate Locations tool beforehand for your origins and destinations, and then it will be able to use the precalculated network location fields anyway for best efficiency.  However, depending on the nature of the problem, you may still encounter another, similar error once the tool attempts to begin the parallelized OD Cost Matrix calculation.
3. Try closing and reopening Pro and maybe cleaning up, deleting, or changing your scratch folder and/or scratch geodatabase.  I've occasionally seen problems with scratch geodatabases getting messed up, and sometimes that causes errors in my tools.
4. If you have a custom Python conda environment activated in Pro, you could try reverting to the default.
5. Close Pro, reboot your machine.

If none of that works, and you're able to share your data with me, I can try to reproduce the problem using your data.  Perhaps it's something specific to your data, which is why I couldn't reproduce it with my own test data.  If you don't want to post your data publicly, you can share it with me privately at mmorang - at - esri - dot - com.

**Comment by mmorang** [comment]
I exchanged several e-mails privately with @AuroraBao and was never able to reproduce the problem she was having.  Aurora, did you ever find a solution, or have you moved on?  I would like to close this issue if it's been resolved or is no longer relevant.
